### PR TITLE
Password Reset

### PR DESCRIPTION
**Pull request by andrewconner**

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/model/PasswordReset.scala [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+case class PasswordReset (
+  id: Option[Id[PasswordReset]] = None,
+  createdAt: DateTime = currentDateTime,
+  updatedAt: DateTime = currentDateTime,
+  userId: Id[User],
+  state: State[PasswordReset] = PasswordResetStates.ACTIVE,
+  token: String,
+  usedAt: Option[DateTime] = None,
+  usedByIP: Option[String] = None,
+  sentTo: Option[String] = None
```
Comment (2is10) [inline kifi-backend/modules/common/app/com/keepit/model/PasswordReset.scala:17]: maybe add TODOs for type safety:

```
usedByIP: Option[String] = None,  // TODO: Option[IPAddress]
sentTo: Option[String] = None  // TODO: Option[EmailAddress]
```

File: kifi-backend/modules/common/conf/evolutions/shoebox/118.sql [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+# --- !Ups
+
+CREATE TABLE password_reset (
+    id bigint(20) NOT NULL AUTO_INCREMENT,
+    created_at datetime NOT NULL,
+    updated_at datetime NOT NULL,
+    user_id bigint(20) NOT NULL,
+    state VARCHAR(20) NOT NULL,
+    token VARCHAR(64) NOT NULL,
```
Comment (2is10) [inline kifi-backend/modules/common/conf/evolutions/shoebox/118.sql:11]: we’ll need an index on the token

File: kifi-backend/modules/common/app/com/keepit/model/PasswordReset.scala (hunk 1 above)
Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/model/PasswordReset.scala:17]: I can make `sentTo` an `EmailAddressHolder`.

**Comment by 2is10** [comment]
Nice!

File: kifi-backend/modules/common/app/com/keepit/model/PasswordReset.scala (hunk 1 above)
Comment (2is10) [inline kifi-backend/modules/common/app/com/keepit/model/PasswordReset.scala:17]: I think we need to make a simple EmailAddress AnyVal type and rename the
model to UserEmailAddress.

Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/model/PasswordReset.scala:17]: We basically do with `EmailAddressHolder`. Agreed on `UserEmailAddress` naming.

**Pull request merged**